src/sentry/api/endpoints/organization_transaction_details.py

```diff
@@ -0,0 +1,314 @@
+from rest_framework.response import Response
+from sentry_relay.consts import SPAN_STATUS_CODE_TO_NAME
+from snuba_sdk import Column
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.constants import ObjectStatus
+from sentry.models.project import Project
+from sentry.search.events.builder.spans_indexed import SpansIndexedQueryBuilder
+from sentry.search.events.types import ParamsType
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.referrer import Referrer
+
+
+@region_silo_endpoint
+class OrganizationTransactionDetailsEndpoint(OrganizationEndpoint):
+    owner = ApiOwner.PERFORMANCE
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+    }
+
+    def get(self, request: Request, organization, project_slug, transaction_id):
+        try:
+            project = Project.objects.get(
+                slug=project_slug, organization_id=organization.id, status=ObjectStatus.ACTIVE
+            )
+        except Project.DoesNotExist:
+            return Response(status=404)
+
+        # Check access to the project as this endpoint doesn't use membership checks done
+        # get_filter_params().
+        if not request.access.has_project_access(project):
+            return Response(status=404)
+
+        # Get a list of all spans with this transaction_id
+        start = request.GET.get("start_timestamp")
+        end = request.GET.get("end_timestamp")
+
+        spans_data = _query_all_spans_in_transaction(
+            organization, project, transaction_id, start, end
+        )
+        if len(spans_data) == 0:
+            return Response(status=404)
+
+        segment_span = next((span for span in spans_data if span["is_segment"] != 0), None)
+        if segment_span is None:
+            return Response({"detail": "No segment span found"}, status=500)
+
+        spans_data = list(
+            filter(lambda span: span["span_id"] != segment_span["span_id"], spans_data)
+        )
+
+        segment_tags_items = zip(segment_span["tags.key"], segment_span["tags.value"])
+        segment_sentry_tags = dict(
+            zip(segment_span["sentry_tags.key"], segment_span["sentry_tags.value"])
+        )
+        segment_measurements = dict(
+            zip(segment_span["measurements.key"], segment_span["measurements.value"])
+        )
+
+        transaction_name = segment_sentry_tags.get("transaction")
+        parent_span_id = segment_span.get("parent_span_id", None)
+        if parent_span_id == "00":
+            parent_span_id = None
+
+        # Assemble them into the same format as the event details endpoint
+        return Response(
+            {
+                "event_id": transaction_id,
+                "project": project.id,
+                "release": segment_sentry_tags.get("release"),
+                "dist": None,
```

Comment:
this being None doesn't seem right -- do we not store this info?

Reply:
Not in any of the test events I've looked at, but I'll take another look at prod.